routine_butler/elements/routines_sidebar.py

```diff
@@ -297,29 +290,31 @@
 #                     ),
 #                 )
 
-#     def add_routine_item(self, session: Session):
-#         logger.debug("Adding routine item")
-#         session.add(self.routine)
-#         # create routine item to routine object with order index of the last_reg_idx + 1
-#         routine_item = RoutineItem(order_index=self.last_reg_idx + 1)
-#         # increment postceding (reward) routine items' order indexes
-#         for ri in self.routine.routine_items:
-#             if ri.order_index > self.last_reg_idx:
-#                 ri.order_index += 1
-#         # insert into list at new index
-#         self.routine.routine_items.append(routine_item)
-#         # update self.last_reg_idx
-#         self.last_reg_idx += 1
-#         logger.debug(f"Last regular routine item idx now: {self.last_reg_idx}")
-#         if any(ri.is_reward for ri in self.routine.routine_items):
-#             # update the frame if an insertion is being made
-#             self._update_items_frame()
-#         else:
-#             # just add the row
-#             with self.items_frame:
-#                 self._add_row(routine_item)
-#         # update db
-#         session.commit()
+    def on_add_routine_item(self):
+        logger.debug("Adding routine item")
+        with self.ctl.session() as session:
+            routine = session.get(Routine, self.routine_id)
```

Comment:
also should be moved into the Controller.
I guess the logic could be simplified... I guess it would be easier to just re-assign the index/order at every change.